### PR TITLE
Improve tests and CI

### PR DESCRIPTION
**Pull request by mike-sul**
- ci: Improve CI checks and tests
  - no docker in docker is needed anymore
  - move most of CI runs to the github workflow
- tests: Improve event sending in the tests
- tests: Split the rollback tests into two test suites
- tests: Make tests independent of the real `dockerd`
- aklite-offline: Add param to inject a custom docker client
- reportqueue: Catch exceptions while loading events from DB

### CODE REVIEW

**Comment by mike-sul** [comment]
[ci-retest] Just checking if there are no any transient failures anymore.

**Comment by mike-sul** [comment]
ci-retest] Just checking if there are no any transient failures anymore.

File: tests/test_lite.sh [hunk 1]
```diff
@@ -18,9 +20,28 @@ cleanup() {
         echo "killing webserver"
         kill $pid
     fi
+    if [ -n "${daemon_pid}" ] ; then
+      echo "killing docker daemon mock: "${daemon_pid}""
+      kill "${daemon_pid}"
```
Comment (doanac) [inline tests/test_lite.sh:25]: you sure this is getting called? I'd think the daemon_pid defined below was local to that function?

**Review by doanac — APPROVED** [review]
that was a lot easier to read. thanks

File: tests/test_lite.sh (hunk 1 above)
Comment (mike-sul) [reply tests/test_lite.sh:25]: Yep, it works.

```
104: + cleanup
104: + echo 'cleaning up temp dir'
104: cleaning up temp dir
104: + rm -rf /tmp/tmp.v4qKr9bzbl
104: + '[' -n 31 ']'
104: + echo 'killing webserver'
104: killing webserver
104: + kill 31
104: + '[' -n 16 ']'
104: + echo 'killing docker daemon mock: 16'
104: killing docker daemon mock: 16
104: + kill 16

```

**Pull request merged**